File: Python/Kode Program/Turnamen.py
from PlayerTCG import PlayerTCG

class Turnamen:
    def __init__(self, id="", nama="", tanggal=""):
        self.id = id
        self.nama = nama
        self.tanggal = tanggal
        self.listPlayer = []

    def setID(self, i): self.id = i
    def setNama(self, n): self.nama = n
    def setTanggal(self, t): self.tanggal = t

    def getID(self): return self.id
    def getNama(self): return self.nama
    def getTanggal(self): return self.tanggal
    def getListPlayer(self): return self.listPlayer

    def addPlayer(self, p: PlayerTCG): self.listPlayer.append(p)

    def getPlayerByID(self, pid: str):
        for p in self.listPlayer:
            if p.getID() == pid:
                return p
        return None

    def printTurnamen(self):
        print(f"Turnamen ID: {self.id}, Nama: {self.nama}, Tanggal: {self.tanggal}")
        print("Daftar Player:")
        for p in self.listPlayer:
            p.printInfo()
            p.printDeck()
            print("-----------------------------")

File: CPP/Kode Program/Turnamen.cpp
#include "PlayerTCG.cpp"
#include <vector>
using namespace std;

class Turnamen {
private:
    string id;
    string nama;
    string tanggal;
    vector<PlayerTCG> listPlayer;

public:
    Turnamen() {}
    Turnamen(string id, string nama, string tanggal)
        : id(id), nama(nama), tanggal(tanggal) {}

    void setID(string i) { id = i; }
    void setNama(string n) { nama = n; }
    void setTanggal(string t) { tanggal = t; }

    string getID() const { return id; }
    string getNama() const { return nama; }
    string getTanggal() const { return tanggal; }
    vector<PlayerTCG>& getListPlayer() { return listPlayer; }

    void addPlayer(const PlayerTCG &p) { listPlayer.push_back(p); }

    PlayerTCG* getPlayerByID(const string &pid) {
        for (auto &p : listPlayer) {
            if (p.getID() == pid) return &p;
        }
        return nullptr;
    }

    void printTurnamen() {
        cout << "Turnamen ID: " << id << ", Nama: " << nama << ", Tanggal: " << tanggal << endl;
        cout << "Daftar Player:\n";
        for (auto &p : listPlayer) {
            p.printInfo();
            p.printDeck();
            cout << "-----------------------------\n";
        }
    }
};
